docs/sources/flow/reference/components/discovery.kubelet.md

```diff
@@ -0,0 +1,145 @@
+---
+title: discovery.kubelet
+---
+
+# discovery.kubelet
+
+`discovery.kubelet` discovers Kubernetes pods running on the specified Kubelet
+and exposes them as scrape targets.
+
+## Usage
+
+```river
+discovery.kubelet "LABEL" {
+  kubelet_url = "KUBELET_URL"
+  bearer_token_file = "TOKEN_FILE"
+}
+```
+
+## Requirements
+
+* The Kubelet must be reachable from the `grafana-agent` pod network.
+* Follow the [Kubelet authorization](https://kubernetes.io/docs/reference/access-authn-authz/kubelet-authn-authz/#kubelet-authorization)
+  documentation to configure authentication to the Kubelet API.
+
+## Arguments
+
+The following arguments are supported:
+
+Name | Type | Description | Default | Required
+---- | ---- | ----------- | ------- | --------
+`kubelet_url` | `string` | URL of Kubernetes API server. | | yes
+`bearer_token` | `secret` | Bearer token to authenticate with. | | no
+`bearer_token_file` | `string` | File containing a bearer token to authenticate with. | | no
+`refresh_interval` | `duration` | How often the Kubelet should be polled for scrape targets | `5s` | no
+`namespaces` | `list(string)` | A list of namespaces to extract target pods from | | no
+
+One of the following authentication methods must be provided if kubelet authentication is enabled
+ - [`bearer_token` argument](#arguments).
+ - [`bearer_token_file` argument](#arguments).
+ - [`authorization` block][authorization].
+
+The `namespaces` list limits the namespaces to discover resources in. If
+omitted, all namespaces are searched.
+
+## Blocks
+
+The following blocks are supported inside the definition of
+`discovery.kubelet`:
+
+Hierarchy | Block | Description | Required
+--------- | ----- | ----------- | --------
+authorization | [authorization][] | Configure generic authorization to the endpoint. | no
+tls_config | [tls_config][] | Configure TLS settings for connecting to the endpoint. | no
+
+[authorization]: #authorization-block
+[tls_config]: #tls_config-block
+
+### authorization block
+
+{{< docs/shared lookup="flow/reference/components/authorization-block.md" source="agent" >}}
+
+### tls_config block
+
+{{< docs/shared lookup="flow/reference/components/tls-config-block.md" source="agent" >}}
+
+## Exported fields
+
+The following fields are exported and can be referenced by other components:
+
+Name | Type | Description
+---- | ---- | -----------
+`targets` | `list(map(string))` | The set of targets discovered from the Kubelet API.
+
+Each target includes the following labels::
+
+* `__meta_kubernetes_namespace`: The namespace of the pod object.
```

Comment:
Should we add  `__address__` should we add it to this list? Also given this is an internal API should there be a disclaimer that these are not guaranteed?

Reply:
I've added the address label to the list, when you say not guaranteed do you mean we should call out these may be subject to change or something else?

Reply:
Have added the address label and added a note around the Kubelet API being internal